### DISCUSSION TITLE
Should we migrate documentation examples to ESM?

### DISCUSSION THREAD

**Discussion by oscard0m** [octokit/octokit.js#2491] [Ideas] [answered]
In the context of migrating to Node 18, would make sense to plan the migration of code examples of `README.md` and other documentation files?

@octokit/js-community @octokit/maintainers 

Context: https://github.com/octokit/request.js/pull/609#issuecomment-1627808164

**Comment by wolfy1339** [comment]
Officially, the main entry point for NodeJS is still CommonJS.
One can still use ESM and it should pick up the right files

**Reply by oscard0m** [reply]
I guess the discussion is if we want to swap the "priority" of ESM over CJS or not in our docs. We can do something like this:

```diff
- const { request } = require("@octokit/request");
- // or: import { request } from "@octokit/request";
+ import { request } from "@octokit/request";
+ // or: const { request } = require("@octokit/request");
```

**Reply by wolfy1339** [reply]
A lot of modules are going that way, simply look through the PRs opened by renovate that are blocked by ESM. I think it's a good call to make

**Reply by gr2m** [reply]
I'm okay either way. Thanks for opening the discussion @oscard0m

**Reply by kfcampbell** [reply, marked as answer]
Perhaps prioritizing ESM would make docs changes ever so slightly quicker when we do convert to ESM fully. I would be good to  ship Oscar's changes in https://github.com/octokit/request.js/pull/609.
